ViewEnvironmentUI/Sources/ViewEnvironmentPropagating.swift

```diff
@@ -105,22 +126,248 @@ extension ViewEnvironmentPropagating {
+    ///
+    /// This describes the ancestor that the `ViewEnvironment` is inherited from.
+    ///
+    /// To override the return value of this property, set the `environmentAncestorOverride`.
+    /// If no override is present, the return value will be `defaultEnvironmentAncestor`.
+    ///
+    @_spi(ViewEnvironmentWiring)
+    public var environmentAncestor: ViewEnvironmentPropagating? {
+        environmentAncestorOverride?() ?? defaultEnvironmentAncestor
+    }
+
+    /// The `ViewEnvironment` propagation descendants.
+    ///
+    /// This describes the descendants that will be notified when the `ViewEnvironment` changes.
+    /// 
+    /// If a descendant's `environmentAncestor` is not `self`, that descendant will not be notified when the
+    /// `ViewEnvironment` changes.
+    ///
+    /// To override the return value of this property, set the `environmentDescendantsOverride`.
+    /// If no override is present, the return value will be `defaultEnvironmentDescendants`.
+    ///
+    @_spi(ViewEnvironmentWiring)
+    public var environmentDescendants: [ViewEnvironmentPropagating] {
+        environmentDescendantsOverride?() ?? defaultEnvironmentDescendants
+    }
+
+    /// ## SeeAlso ##
+    /// - `environmentAncestorOverride`
+    ///
+    @_spi(ViewEnvironmentWiring)
+    public typealias EnvironmentAncestorProvider = () -> ViewEnvironmentPropagating?
+
+    /// This property allows you to override the propagation path of the `ViewEnvironment` as it flows through the
+    /// node hierarchy by overriding the return value of `environmentAncestor`.
+    ///
+    /// The result of this closure should typically be the propagation node that the `ViewEnvironment` is inherited
+    /// from. If the value of the ancestor is nil, by default, other nodes configured with this node as a descendant
+    /// will not notify this node of needing an environment update as it changes. This allows a node to effectively 
+    /// act as a root node when needed (e.g. bridging from other propagation systems like WorkflowUI).
+    ///
+    /// If this value is `nil` (the default), the resolved value for the ancestor will be `defaultEnvironmentAncestor`.
+    ///
+    /// ## Important ##
+    ///  - You must not set overrides while overrides are already set—doing so will throw an assertion. This assertion
+    ///    prevents accidentally clobbering an existing propagation path customization defined somewhere out of your
+    ///    control (e.g. Modals customization).
+    ///
+    @_spi(ViewEnvironmentWiring)
+    public var environmentAncestorOverride: EnvironmentAncestorProvider? {
+        get {
+            objc_getAssociatedObject(self, &AssociatedKeys.ancestorOverride) as? EnvironmentAncestorProvider
+        }
+        set {
+            assert(
+                newValue == nil
+                    || environmentAncestorOverride == nil,
+                "Attempting to set environment ancestor override when one is already set."
```

Comment:
IMO it's a bit odd to require clearing a setter before you set it again. Thoughts?

Reply:
Yup, it's definitely a bit weird, but we had it in place to ensure folks aren't overriding an existing override without knowing it—this assertion has caught unexpected scenarios like this in the past. 

Now that the framework has evolved a bit more, and I think we're more confident in the right shape for how to utilize these types, I don't feel as strongly that it's still necessary. I'm fine with removing it now if other folks agree!

Reply:
+1. Is it intended that we allow this to be `nil`'d out?

Reply:
I'm in favor of leaving it. If the propagation path gets messed up it can cause very subtle bugs downstream that are difficult to diagnose. I think the extra step to unwire existing overrides before wiring new ones is worth it to avoid that kind of mistake.

Reply:
OK, I'm leaning towards keeping it for now, but let me know if anyone feels strongly that this shouldn't require unwiring before overriding!